core/win32/syscall.c

```diff
@@ -2747,25 +2747,28 @@ pre_system_call(dcontext_t *dcontext)
     DOLOG(2, LOG_SYSCALLS, {
           dump_mcontext(mc, THREAD, false/*not xml*/);
     });
-    /* we can't pass other than a numeric literal anymore */
-    LOG(THREAD, LOG_SYSCALLS, 3, "\tparam 0: "PFX"\n",
-        sys_param(dcontext, param_base, 0));
-    LOG(THREAD, LOG_SYSCALLS, 3, "\tparam 1: "PFX"\n",
-        sys_param(dcontext, param_base, 1));
-    LOG(THREAD, LOG_SYSCALLS, 3, "\tparam 2: "PFX"\n",
-        sys_param(dcontext, param_base, 2));
-    LOG(THREAD, LOG_SYSCALLS, 3, "\tparam 3: "PFX"\n",
-        sys_param(dcontext, param_base, 3));
-    LOG(THREAD, LOG_SYSCALLS, 3, "\tparam 4: "PFX"\n",
-        sys_param(dcontext, param_base, 4));
-    LOG(THREAD, LOG_SYSCALLS, 3, "\tparam 5: "PFX"\n",
-        sys_param(dcontext, param_base, 5));
-    LOG(THREAD, LOG_SYSCALLS, 3, "\tparam 6: "PFX"\n",
-        sys_param(dcontext, param_base, 6));
-    LOG(THREAD, LOG_SYSCALLS, 3, "\tparam 7: "PFX"\n",
-        sys_param(dcontext, param_base, 7));
-    LOG(THREAD, LOG_SYSCALLS, 3, "\tparam 8: "PFX"\n",
-        sys_param(dcontext, param_base, 8));
+    /* Needs to check readability before logging. */
```

Comment:
Check is needed because the program can set param_base to an invalid value (unreadable address in memory) and that happens with the test case

Reply:
Sure, though this method is racy as the memory could become unreadable concurrently with this code.  The more robust method is a safe_read or try/except, but since this is just logging code this is ok if it makes the code simpler: but please use a comment stating we're aware of the race.

Reply:
ok